### DISCUSSION TITLE
[SUGGESTION] Misc Part Requests

### DISCUSSION THREAD

**Discussion by tristenbarrett** [tsunamayo/Starship-EVO#4024] [Suggestions]
Hello,

As I've spent time getting to know the game recently I've come across a few bits and pieces I'd like to see that I feel would enhance the current lineup quickly and easily:

1. Pipeline T-pieces.  While even more join types would be great, as long as there are t-pieces, there will always be a way to make most possible join types.
2. Hull blocks with the Bevel shape
3. Variant of the screen block measuring 4w,1h,1d using the same centred depth as the current part.  (Think hanging sign from the roof of a hallway).  Ideal if the mount point is on top as I suspect it will usually be used hanging.
4. The same but aligned to the rear of the depth for wall mounting (think sign above a door)
5. A variant of the Button Wedged designed for mounting side-on rather than upright.
6. A wedged panel piece like the Indicator Light but with the light also on the diagonal face.  Light to be small and subtle (for status indication up close without being distracting at a distance.
7. A clear version of the small frosted windows.  I know this sort of exists as the 'Window Tile' but that part has a different minimum size and borders.
8. A 'Thin Tile' with some sort of edge on one side.  Either make the whole thing a wedge, or add a slightly bevelled edge just on one side (as opposed to the Panel Bevelled which has it on all sides).
9. Pipeline 'airtight' piece.  A single cube of pipe which has a surrounding bracket, allowing it to form an airtight wall when surrounded with cubes.  Basically a wall pass-through if that makes sense?
10. A pipe piece with a valve (only decorative for now of course!)
11. Lots more decals!  There's a good library of symbols in the game already (such as sprites 0 through 60 from the screens) which could be implemented.  Two additional things that also spring to mind are a line (to match with the 'rounded corner') and a lock/unlock symbol.  And of course at some stage letters and numbers would be amazing!

Many thanks!

**Comment by Kaiser-Indrasil** [comment]
I agree with most, with the exception of these few:
2. You can make your own "blocks" with bricks. Yes, it wouldn't be airtight but implementing several half-height block variants would be very hard, especially in the voxel system, I imagine. The bevel alone doesn't have its own individual collision box and I think it's because it would just be taxing on the calculations.
4. You can offset it using tiles and thin tiles.
9. That would just make it a regular armour block or brick. A pipe going straight into a wall seems perfectly reasonable to me.
10. A simple T-juction with a thin round tile and some stickers would do the trick.

Actually, I want to expand upon the decals. Seeing as they would eventually be compatible with the copy-paste system, it would be cool if we could save them as blueprints and upload them onto the Workshop for people to stamp onto their creations.

Forza Horizon 4 has this kinda thing and it massively streamlines the painting process. Of course, it also handles wrapping which, I assume, wouldn't be feasible for a game like Stevo, but being able to paste a decal composition onto a simple flat surface? Would be very nice.

**Comment by tristenbarrett** [comment]
Really appreciate the feedback!  For clarity, my thinking on a couple of points was as follows...

2) By bevel I don't mean one of the half height blocks, but rather the full size version of the Construction block called Bevel.  It's the one that's basically a wedge for 50% of 2 axis.

4) The screen is a bit weird as it's actually offset from the half/quarter parts a bit by being centred on the small grid.  While it seems to be the 25%-of-a-block measurement like a Thin Tile, because it's centred it actually take up 50% of the space with gaps before and after.  Actually, maybe instead of #3 and #4 it would be better just to get it inline with the Thin Tile piece and then as you say, we can adjust it as we need to.

<img width="310" alt="Screen Centering" src="https://user-images.githubusercontent.com/24539048/106729657-a5f37e80-6605-11eb-92e2-e0294a9eefc6.png">

9) and 10) are just on the off chance that one day pipes become more than just decorative.  I'm already building with Airflow, plumbing and fuel flow in mind and even if it never becomes a thing, I'm still enjoying building with that level of fidelity.

**Comment by Garrett-C** [comment]
One point about your point 3. I don't think it should mount at the top. It would be the only block to do that which isn't very intuitive in my opinion. 

In fact thinking about it I am pretty sure all blocks just mount based on what face you look at.

**Comment by tristenbarrett** [comment]
> One point about your point 3. I don't think it should mount at the top. It would be the only block to do that which isn't very intuitive in my opinion.

Yeup, I'd agree with that.  I've now further revised my original opinion and feel instead of 3 and 4, we just need one aligned to the 'Thin Tile' 1/4 of a block style, in which case the likely mount point would be the back.  In any case I guess in Evo it doesn't matter too much as you can always build a frame to get the part in place then remove.
